src/simulation_node.py

```diff
@@ -167,38 +141,78 @@ def subscribing_control_topic(self):
-
-            # Create collision and visual shapes for the link
-            collision_shape_id = p.createCollisionShape(p.GEOM_BOX, halfExtents=[x / 2 for x in link_params['dimensions']])
-            visual_shape_id = p.createVisualShape(p.GEOM_BOX, halfExtents=[x / 2 for x in link_params['dimensions']])
             
-            # Create the multibody for the link
-            body_id = p.createMultiBody(baseMass=link_params['mass'],
-                                        baseCollisionShapeIndex=collision_shape_id,
-                                        baseVisualShapeIndex=visual_shape_id,
-                                        basePosition=link_params.get('position', [0, 0, 0]),  
-                                        baseInertialFramePosition=link_params.get('inertia_position', [0, 0, 0]),
-                                        physicsClientId=client_id)
-            
-            # Set the dynamics properties for the link
-            p.changeDynamics(body_id, -1,
-                            restitution=link_dynamics['restitution'],
-                            lateralFriction=link_dynamics['lateralFriction'],
-                            spinningFriction=link_dynamics['spinningFriction'],
-                            rollingFriction=link_dynamics['rollingFriction'],
-                            contactDamping=link_dynamics['contactDamping'],
-                            contactStiffness=link_dynamics['contactStiffness'],
-                            physicsClientId=client_id)
-            
-            # Store the body ID for later use (e.g., control or simulation)
-            self.robots.append(body_id)
+        """Create the robot using dynamics and structure settings with joints, using a mesh for the pedestal."""
+        p.loadURDF("plane.urdf", physicsClientId=client_id)
+
+        # Create the collision and visual shapes for world_box and pedestal
+        shapes = {}
+        for link_name, link_config in self.structure_config.items():
+            shapes[link_name] = {
+                'collision': p.createCollisionShape(p.GEOM_BOX, halfExtents=[x / 2 for x in link_config['dimensions']]),
+                'visual': p.createVisualShape(p.GEOM_BOX, halfExtents=[x / 2 for x in link_config['dimensions']])
+            }
+
+        # Fetch the mesh path for the pedestal from ROS2 parameters
+        pedestal_mesh_path = self.get_parameter('structure.pedestal.mesh').get_parameter_value().string_value
+
+        # Load the mesh for the pedestal
+        shapes['pedestal'] = {
+            'collision': p.createCollisionShape(shapeType=p.GEOM_MESH, fileName=pedestal_mesh_path, meshScale=[0.5, 0.5, 0.5]),
+            'visual': p.createVisualShape(shapeType=p.GEOM_MESH, fileName=pedestal_mesh_path, meshScale=[0.5, 0.5, 0.5])
+        }
+
+        # Create the base (world_box)
+        world_box_pos = [0, 0, self.structure_config['world_box']['dimensions'][2] / 2]
+        base_id = p.createMultiBody(
```

Comment:
you should add joint types in createMultiBody. See this tutorial: https://github.com/bulletphysics/bullet3/blob/master/examples/pybullet/examples/createMultiBodyLinks.py